### ISSUE TITLE
DataDog actor monitoring that was working with Phobos.Actor 1.2.5 now not working after upgrading to 1.4.2

### ISSUE THREAD

**Issue by andrew-lewell** [petabridge/phobos-issues#53]
Hi, I have set up DataDog monitors to trigger alerts for actor restarts and it was working fine with 1.2.5 but after upgrading Phobos.Actor to 1.4.2 it does not seem to trigger them anymore. Any ideas what could be the issue here? Has something changed around the Akka.NET.akka.actor.restarts.counter.value?

**Comment by Aaronontheweb** [comment]
Off the top of my head, the only Phobos-specific thing I can think of that may have had an impact was when we added actor lifecycle tracing in 1.4 - but that change shouldn't have affected any metrics around actor restarts: https://phobos.petabridge.com/articles/releases/RELEASE_NOTES.html#140httpssdkbincompublisherpetabridgeproductphobospackagesphobosactorversions140-january-19th-2022

**Comment by Aaronontheweb** [comment]
Looks like Phobos 1.4 depends on App.Metrics 4.3.0 https://sdkbin.com/publisher/petabridge/product/phobos/packages/phobos.monitoring/versions/1.4.2

https://sdkbin.com/publisher/petabridge/product/phobos/packages/phobos.monitoring/versions/1.2.5 - 1.2.5 used App.Metrics 4.2.0

There were a number of changes to the StatsD packages between 4.2.0 and 4.3.0 - are you using Phobos with the App.Metrics support for DogStatsD by any chance?

https://github.com/AppMetrics/AppMetrics/commits/features/4.3.0

**Comment by Aaronontheweb** [comment]
Could possibly be related to https://github.com/AppMetrics/AppMetrics/issues/682 - but we'd need to know which versions of App.Metrics' packages for DataDog you're using

**Comment by andrew-lewell** [comment]
I am using App.Metrics 4.3.0. 

I'll take another look at the metric name, perhaps it has changed. Thanks for the tip.

**Comment by Aaronontheweb** [comment]
> I am using App.Metrics 4.3.0.
> 
> I'll take another look at the metric name, perhaps it has changed. Thanks for the tip.

Did you upgrade from an earlier version of App.Metrics as part of this? If that's the case then the metric name is probably the best guess.

**Comment by Aaronontheweb** [comment]
@jackowild let me take a look and see if we did anything - because that is _very_ odd. We haven't really touched the metrics code that much in the 1.x branch recently.

**Comment by Aaronontheweb** [comment]
Welp, I found the problem @jackowild - from the internals of Phobos.Metrics:

![image](https://user-images.githubusercontent.com/326939/163010676-eeb85843-8497-44bd-8e70-f05d0f6ef53e.png)

We fixed a small `NullReferenceException` that could happen with App.Metrics but apparently did it in such a way that overwrote all GlobalTags previously set by the user. This happened in Phobos 1.4.1.

We're about to ship Phobos 2.0 and 1.5 today - I'll add a regression text for 1.5 for this use case and fix it there before we publish.
